src/query/service/src/interpreters/interpreter.rs

```diff
@@ -228,10 +234,31 @@ fn log_query_finished(ctx: &QueryContext, error: Option<ErrorCode>, has_profiles
 pub async fn interpreter_plan_sql(ctx: Arc<QueryContext>, sql: &str) -> Result<(Plan, PlanExtras)> {
     let mut planner = Planner::new(ctx.clone());
     let result = planner.plan_sql(sql).await;
+    let short_sql = short_sql(sql.to_string());
+
+    if let Ok((_, extras)) = &result {
+        let mut stmt = extras.statement.clone();
+        if let Statement::Query(_) = stmt {
+            let query_hash = format!("{:x}", Md5::digest(stmt.to_string()));
+            #[derive(VisitorMut)]
+            #[visitor(Literal(enter))]
+            struct AstVisitor {}
+
+            impl AstVisitor {
+                fn enter_literal(&mut self, lit: &mut Literal) {
+                    *lit = Literal::Null;
+                }
+            }
 
-    if result.is_err() {
+            let mut visitor = AstVisitor {};
+            stmt.drive_mut(&mut visitor);
```

Comment:
This code would be better moved to a separate function implementation, which would look cleaner, add some comments by the way will make it more readable.